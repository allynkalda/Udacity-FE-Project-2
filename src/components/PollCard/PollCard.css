.poll-card {
    padding: 1rem;
    margin: 1rem;
    border-radius: 4px;
    text-align: center;
    min-width: 25%;
    background-color: rgb(88, 88, 88);
    color: #fff;
}

.link {
    text-decoration: none;
}