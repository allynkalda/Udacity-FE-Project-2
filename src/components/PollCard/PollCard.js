import { Link } from "react-router-dom";
import { Button } from '@mui/material';

import { formatDate } from '../../utils/date'
import './PollCard.css'

const PollCard = ({ question }) => {

    return (
        <div className="poll-card">
            {!question.author || !question.timestamp || !question.id ? (
                <p data-testid="warning">We cannot find the question</p>
            ) : (
                <>
                <p data-testid="author">{question.author}</p>
                <p data-testid="timestamp">{formatDate(question.timestamp)}</p>
                <Link data-testid="link" to={`/questions/${question.id}`} className="link">
                    <Button sx={{ backgroundColor: 'white', color: '#000', margin: 2}}>Show</Button>
                </Link>
                </>
            )}
        </div>
    )
}

export default PollCard