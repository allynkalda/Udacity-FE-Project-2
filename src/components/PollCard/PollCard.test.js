import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import PollCard from './PollCard';

const mockQuestion = {
  author: 'Author',
  timestamp: 1468479767190,
  id: 123
}

describe('Match snapshot of PollCard', () => {
  it('will match snapshot', () => {
    const view = render(<PollCard question={mockQuestion} />, {wrapper: MemoryRouter})
    expect(view).toMatchSnapshot()
  })
})

test('renders author and button label', () => {

  render(<PollCard question={mockQuestion} />, {wrapper: MemoryRouter})

  expect(screen.getByTestId("author")).toHaveTextContent(mockQuestion.author)
  expect(screen.getByRole("button")).toHaveTextContent("Show")
});

test('renders converted timestamp', () => {

  render(<PollCard question={mockQuestion} />, {wrapper: MemoryRouter})

  expect(screen.getByTestId("timestamp")).toHaveTextContent('9:02:AM | 14/07/2016')
});

test('renders error warning when props is empty', () => {

  render(<PollCard question={{}} />, {wrapper: MemoryRouter})

  expect(screen.getByTestId("warning")).toHaveTextContent('We cannot find the question')
});

test('renders error warning when props has single element', () => {

  render(<PollCard question={{ author: 'Me' }} />, {wrapper: MemoryRouter})

  expect(screen.getByTestId("warning")).toHaveTextContent('We cannot find the question')
});

test('link goes to a different route', async () => {
  const history = createMemoryHistory();

  history.push = jest.fn();

  render(
    <Router location={history.location} navigator={history}>
      <PollCard question={mockQuestion} />
    </Router>
  )

  const link = screen.getByTestId('link')
  fireEvent.click(link);

  expect(history.push).toHaveBeenCalledWith(    {
    hash: '',
    pathname: '/questions/123',
    search: '',
  },
  undefined,
  {"preventScrollReset": undefined, "relative": undefined, "replace": false, "state": undefined}
  );
});