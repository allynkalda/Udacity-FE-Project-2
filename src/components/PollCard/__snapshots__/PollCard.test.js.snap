// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot of PollCard will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="poll-card"
      >
        <p
          data-testid="author"
        >
          Author
        </p>
        <p
          data-testid="timestamp"
        >
          9:02:AM | 14/07/2016
        </p>
        <a
          class="link"
          data-testid="link"
          href="/questions/123"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-fakkgz-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Show
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="poll-card"
    >
      <p
        data-testid="author"
      >
        Author
      </p>
      <p
        data-testid="timestamp"
      >
        9:02:AM | 14/07/2016
      </p>
      <a
        class="link"
        data-testid="link"
        href="/questions/123"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-fakkgz-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Show
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
